tests/integration/test_manifest_submission.py

```diff
@@ -0,0 +1,97 @@
+import logging
+import os
+import shutil
+
+import pandas as pd
+import pytest
+import requests
+from synapseclient.client import Synapse
+
+from schematic.utils.general import create_temp_folder
+from tests.utils import CleanupItem
+
+logging.basicConfig(level=logging.DEBUG)
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope="session")
+def download_location():
+    temporary_manifest_storage = "temporary_manifest_storage"
+    if not os.path.exists(temporary_manifest_storage):
+        os.makedirs(temporary_manifest_storage)
+    full_parent_path = os.path.abspath(temporary_manifest_storage)
+
+    download_location = create_temp_folder(full_parent_path)
+    yield download_location
+
+    # Cleanup after tests have used the temp folder
+    if os.path.exists(download_location):
+        shutil.rmtree(download_location)
```

Comment:
@linglp 
Should this go into conftest.py so it can be re-used? The name of the storage location doesn't matter quite so much, so it could be anything.

Also to note, a session scope with a temp folder that does not have a unique name is problematic. The reason is that when xdist runs tests it starts up new processes to run the code. Each process has a unique session scope that is not shared. When you use a non-unique folder to store content, which later gets cleaned up, it is introducing a race condition in the code. If the same worker picks up each tests that uses this location generator - Great no issues. If different workers pick up tests that share this, whoever finishes the test first is going to delete the directory tree and break the test that might still be running.

https://github.com/Sage-Bionetworks/schematic/pull/1505/files#diff-664ac28bfa7dd824a516813cb7f560aa7207c6a7bb54b7053cc9dcfd9c157d46 includes the changes I had introduced to solve this non-unique problem (However, that code has not been merged)

Reply:
Thanks for the feedback @BryanFauble . I updated to move download_location fixture to conftest and use uuid as folder name. Does that work?

Reply:
> I updated to move download_location fixture to conftest

This is great!

> use uuid as folder name. Does that work?

This does not work. I added a commit to this branch, take a look: https://github.com/Sage-Bionetworks/schematic/pull/1512/commits/9fc7fd1b2d1d9b33f8764f218c4984a0ceac2c7e


In the end out goal is that these temporary directories are being make in the temp directory for the applicable OS that is running the test. The solution that you proposed was creating the temp directory at the working directory where the code was being ran.

Reply:
I see. Thanks for explaining Bryan. To make sure that I understand, the difference is that tempfile.gettempdir() would create temporary folders in the system's temp directory and my approach was creating unique folders at the working directory?

Reply:
Yes, that's correct @linglp

Reply:
Thanks Bryan. I will mark this comment as resolved :)